src/routes/index.js

```diff
@@ -1,24 +1,43 @@
-import React from 'react'
-import { Route } from 'react-router-dom'
+import React, { useEffect } from 'react'
+import { Route, useHistory, Switch } from 'react-router-dom'
 import Question from '../views/Question'
 import Home from '../views/Home'
 import Login from '../views/Login'
 import Signup from '../views/Signup'
 import AskQuestion from '../views/AskQuestion'
 import About from '../views/About'
 import NotFound from '../views/404'
+import ReactGA from 'react-ga'
 
-export default [
-  <Route
-    exact
-    key='question'
-    path='/question/:id'
-    component={Question}
-  />,
-  <Route exact key='home' path='/' component={Home} />,
-  <Route exact key='login' path='/login' component={Login} />,
-  <Route exact key='signup' path='/signup' component={Signup} />,
-  <Route exact key='ask-a-question' path='/ask-a-question' component={AskQuestion} />,
-  <Route exact key='about' path='/about' component={About} />,
-  <Route key='404' path='*' component={NotFound} />
-]
+ReactGA.initialize('UA-204918741-1')
+
+export default function Routes () {
+  const history = useHistory()
+  useEffect(() => {
+    // To not track activities in GA during development testing
+    if (window.location.hostname !== 'localhost') {
+      trackPageView() // To track the first pageview upon load
+      history.listen(trackPageView) // To track the subsequent pageviews
```

Comment:
Do you need to stop listening if the component is unmounted?

Reply:
yes, it's a good practice to stop listening so that we don't take up so much computing power. We can do that by returning the history.listen.

reference: https://www.hannaliebl.com/blog/use-react-routers-history-listener-with-react-hooks/